ocean_lib/web3_internal/contract_base.py

```diff
@@ -210,7 +215,19 @@ def send_transaction(
             _transact.update(transact)
 
         # TODO: transfer rest of custom contract function, waiting etc.
-        return getattr(self.contract, fn_name)(*fn_args, _transact).txid
+        tx_id = getattr(self.contract, fn_name)(*fn_args, _transact).txid
+        tx_receipt = wait_for_transaction_receipt_and_block_confirmations(
+            web3=self.web3,
+            tx_hash=tx_id,
+            block_confirmations=from_wallet.block_confirmations.value,
+            block_number_poll_interval=BLOCK_NUMBER_POLL_INTERVAL[
+                self.web3.eth.chain_id
+            ],
+        )
+        if tx_receipt.status != 1:
```

Comment:
I think we can either add a timeout option to the call (there is something in constants that defines a max timeout to wait. Do not reject the transaction if unsuccessful, do a timeout-based poll to see when it succeeds.